File: easy/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def findMaxDepth(self, root):
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 1
        left_length = self.findMaxDepth(root.left)
        right_length = self.findMaxDepth(root.right)

        return max(left_length, right_length) + 1

    def maxDepth(self, root: TreeNode) -> int:
        return self.findMaxDepth(root)

File: easy/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
 var maxDepth = function(root) {
    return maxDepthResults(root);
};

const maxDepthResults = function(root){
    if(!root){
        return 0;
    }
    if(!root.left && !root.right){
        return 1
    }

    const maxLeft = maxDepth(root.left);
    const maxRight = maxDepth(root.right);

    return Math.max(maxLeft, maxRight) + 1;
};